content/5_Plat/VTree.mdx

```diff
@@ -0,0 +1,242 @@
+---
+id: VT
+title: Virtual Tree
+author: Ryan Fu
+prerequisites:
+description: Compressing a tree to only necessary nodes
+---
+
+<Resources>
+	<Resource
+		source="YouTube"
+		title="CP Tutorial: Virtual/Auxiliary Tree"
+		url="https://www.youtube.com/watch?v=czySm7bUHgY"
+	>
+	</Resource>
+	<Resource
+		source="CF"
+		title="Virtual trees method"
+		url="https://codeforces.com/blog/entry/140066"
+	>
+	</Resource>
+	<Resource
+		source="OIWiki"
+		title="OIWiki - Virtual Tree"
+		url="https://oi-wiki.org/graph/virtual-tree/"
+	>
+	</Resource>
+	<Resource
+		source="YKW"
+		title="YouKn0wWho Academy - Virtual Tree / Auxiliary Tree"
+		url="https://youkn0wwho.academy/topic-list/virtual_tree"
+	>
+	</Resource>
+
+</Resources>
+
+Often, when running operations or dynamic programming on trees, we only need to keep track of
+a few key nodes and their relationships. If we are able to isolate the few relevant nodes that keep
+the structure of the tree intact, while still maintaining the relationships between key nodes, we can
+cut our time complexity down significantly.
+
+## Example - Unique Occurences
+
+<FocusProblem problem="uniqueocc" />
+
+**Statement:** A tree with $n$ vertices has a color $1 \leq x_e \leq n$ on edge $e$. Let $f(u,v)$ be the number of colors that appear exactly once on that path from $u$ to $v$. Find the sum of $f(u,v)$ across all pairs of vertices $u,v$.
+
+**Constraints:** $2 \leq n \leq 5\cdot 10^5$
+
+### Naive Solution
+
+Instead of counting $f(u,v)$ for each $u,v$, it seems more conducive to count, for each color, the number of paths that contain that color exactly once. We can count this by iterating over each color $c$, and maintaining $\text{dp}[v][0/1]$ as the number of paths up to $v$ which contain the color $c$ $0$ or $1$ times. Our transitions are as follows:
+
+- If our the parent edge of $v$ is not color $c$, we can continue all paths from $v$ upwards, and start a new path from $v$:
+$$
+\texttt{dp}[p][0] \mathrel{+}= \texttt{dp}[v][0] + 1
+$$
+$$
+\texttt{dp}[p][1] \mathrel{+}= \texttt{dp}[v][1]
+$$
```

Comment:
put these two in one tex block and use `\\`

Reply:
it's fine as is